whisper/test.py

```diff
@@ -62,7 +60,7 @@ def test_fp16(self):
         dims = mlx_model.dims
         mels = mx.array(np.random.randn(1, 3_000, dims.n_mels), mx.float16)
         tokens = mx.array(np.random.randint(0, dims.n_vocab, (1, 20)), mx.int32)
-        logits = mlx_model(mels, tokens)
+        logits = mlx_model(mels, tokens).astype(mx.float16)
```

Comment:
Casting to fp16 here defeats the point of the test which is to make sure the enitre inference is in fp16... the test failed for you when you run it without this?

Reply:
Ah, I think the problem is you are using an old mlx which was incorrectly upcasting, we will put out a new release today and see if that fixes the issue.